llmfoundry/eval/datasets/in_context_learning_evaluation.py

```diff
@@ -519,46 +526,12 @@ def collate_fn(self, data: List[Dict[str, Any]]) -> Dict[str, Any]:
         batch['attention_mask'] = ~(batch['input_ids'] == self.pad_tok_id)
         return batch
 
-    def split_batch(self, batch: Any,
-                    microbatch_size: Union[int, float]) -> Sequence[Any]:
-        """Handling for certain specialty columns that must be split into.
-
-        batches in different formats.
-
-        Args:
-            batch (Dict): Batch of data
-            microbatch_size (int | float): Size of microbatches
-
-        Returns:
-            List: List of chunked batches
-        """
-        # Don't split kwargs that don't change
-        # Normally split torch tensors
-        # List split lists of strings
-        if isinstance(microbatch_size, float):
-            raise ValueError(
-                'split_batch does not support floating point microbatch_size.',
-            )
-        chunked = {}
-        for k, v in batch.items():
-            if k in self.static_keys:
-                # Defer broadcasting until we know num_chunks
-                pass
-            elif k in self.list_keys:
-                chunked[k] = _split_list(v, microbatch_size)
-            elif k in self.tensor_keys:
-                chunked[k] = _default_split_batch(v, microbatch_size)
-            else:
-                raise ValueError(f'Unexpected key {k} in batch splitting')
-        num_chunks = len(chunked['input_ids'])
-        for k, v in batch.items():
-            if k in self.static_keys:
-                chunked[k] = [v] * num_chunks
-
-        batched_list = [{k: v[idx]
-                         for k, v in chunked.items()}
-                        for idx in range(num_chunks)]
-        return batched_list
+    def split_batch(
+        self, 
+        batch: Any,
+        microbatch_size: Union[int, float]
+    ) -> Sequence[Any]:
+        return _default_split_batch(batch, microbatch_size)
```

Comment:
This was the resulting behavior of the previous code https://github.com/mosaicml/llm-foundry/blob/5571101a50804406ef0fe23e7ea6795b3c4a1bcb/llmfoundry/eval/datasets/in_context_learning_evaluation.py#L1408-L1417

So I moved the default split_batch function to `InContextLearningGenerationTaskWithAnswersDataset` since it was only being used there and retained the `split_batch` implementation of `InContextLearningMultipleChoiceTaskDataset`. 

Setting split_batch=None internally gets translated to `default_split_batch` here: https://github.com/mosaicml/composer/blob/7fbac5f7f3144ee6c0f7731b125cd308546bc237/composer/core/data_spec.py#L193

Reply:
ahh ok, I follow you now. Makes sense, thanks for explaining!